staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -697,6 +698,9 @@ func shouldSyncUpdatedNode(oldNode, newNode *v1.Node) bool {
 	if oldNode.Spec.ProviderID != newNode.Spec.ProviderID {
 		return true
 	}
+	if oldNode.UID != newNode.UID {
+		return true
+	}
```

Comment:
as @alexanderConstantinescu correctly points out, `shouldSyncUpdatedNode` is only used in the UpdateFunc of the node informer event handler, and since the field is inmutable, this can never happen here